### PR TITLE
[Databox] `az databox job create`: Add new parameter `--transfer-type` `--transfer-configuration-type` `--transfer-filter-details` to support managing import or export jobs

### PR DESCRIPTION
**Pull request by Jing-song**
---
* `az databox job create`: Add new parameter `--transfer-type` `--transfer-configuration-type` `--transfer-filter-details` to support managing import or export` jobs
* `az databox job create/update`: Add new parameter `--kek-type` `--kek-identity` `--kek-url` `--kek-vault-resource-id` to support managing customer managed key
* `az databox job`: Add new subcommand `mark-devices-shipped` to support marking devices shipped
* Migrate `az databox` commands to aaz

close https://github.com/Azure/azure-cli/issues/25778
close https://github.com/Azure/azure-cli/issues/25777
aaz model: https://github.com/Azure/aaz/pull/193

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Databox

File: src/databox/azext_databox/aaz/latest/databox/job/_cancel.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+@register_command(
+    "databox job cancel",
+    confirmation="Are you sure you want to perform this operation?",
+)
+class Cancel(AAZCommand):
+    """CancelJob.
+
+    :example: Cancel job
+        az databox job cancel -g rg --job-name name --reason reason -y
```
Comment (kairu-ms) [inline src/databox/azext_databox/aaz/latest/databox/job/_cancel.py:22]: Usually we will not add `-y` in example to bypass confirmation.

File: src/databox/azext_databox/aaz/latest/databox/job/_delete.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+@register_command(
+    "databox job delete",
+    confirmation="Are you sure you want to perform this operation?",
+)
+class Delete(AAZCommand):
+    """Delete a job.
+
+    :example: Delete job
+        az databox job delete -n job-name -g rg -y
```
Comment (kairu-ms) [inline src/databox/azext_databox/aaz/latest/databox/job/_delete.py:22]: Same problem here.

File: src/databox/azext_databox/azext_metadata.json [hunk 3]
```diff
@@ -1,3 +1,4 @@
 {
-    "azext.minCliCoreVersion": "2.3.1"
+    "azext.isPreview": true,
```
Comment (kairu-ms) [inline src/databox/azext_databox/azext_metadata.json:2]: The previous version is not preview. Please remove it.

File: src/databox/azext_databox/custom.py [hunk 4]
```diff
@@ -1,135 +1,125 @@
...
+                            'include': {
+                                'data_account_type': data_account_type,
+                                'transfer_all_blobs': True,
+                                'transfer_all_files': True
+                            }
+                        }
+                    }
+                    detail['transfer_configuration'] = transfer_configuration
+                if args.transfer_configuration_type == 'TransferUsingFilter':
+                    if not has_value(args.transfer_configuration_type):
```
Comment (kairu-ms) [inline src/databox/azext_databox/custom.py:118]:
```suggestion
                    if not has_value(args.transfer_filter_details):
```

File: src/databox/setup.py [hunk 5]
```diff
@@ -1,20 +1,14 @@
...
-try:
-    from azure_bdist_wheel import cmdclass
-except ImportError:
-    from distutils import log as logger
-    logger.warn("Wheel is not available, disabling bdist_wheel hook")
 
-# TODO: Confirm this is the right version number you want and it matches your
+
 # HISTORY.rst entry.
 VERSION = '0.1.3'
```
Comment (kairu-ms) [inline src/databox/setup.py:13]: Please update VERSION

File: src/databox/HISTORY.rst [hunk 6]
```diff
@@ -3,6 +3,13 @@
 Release History
 ===============
 
+0.1.4
+++++++
+* `az databox job create`: Add new parameter `--transfer-type` `--transfer-configuration-type` `--transfer-filter-details` `--data-box-customer-disk` to support managing import or export` jobs
+* `az databox job create/update`: Add new parameter `--kek-type` `--kek-identity` `--kek-url` `--kek-vault-resource-id` to support managing customer managed key
+* `az databox job`: Add new subcommand `mark-devices-shipped` to support marking devices shipped
+* Migrate `az databox` commands to aaz
```
Comment (jsntcy) [inline src/databox/HISTORY.rst:11]: Remove "Migrate `az databox` commands to aaz" as it's just an internal implementation detail.

File: src/databox/setup.py [hunk 7]
```diff
@@ -1,22 +1,16 @@
...
-    from azure_bdist_wheel import cmdclass
-except ImportError:
-    from distutils import log as logger
-    logger.warn("Wheel is not available, disabling bdist_wheel hook")
 
-# TODO: Confirm this is the right version number you want and it matches your
+
 # HISTORY.rst entry.
-VERSION = '0.1.3'
+VERSION = '0.1.4'
```
Comment (jsntcy) [inline src/databox/setup.py:13]: Let's change the version to `1.0.0` as the first GA.

File: src/databox/HISTORY.rst [hunk 8]
```diff
@@ -3,6 +3,13 @@
 Release History
 ===============
 
+0.1.4
```
Comment (jsntcy) [inline src/databox/HISTORY.rst:6]: Change version number to `1.0.0` as well.

**Review by jsntcy — APPROVED** [review]

**Pull request merged**